### PR TITLE
executor: ignore overflow error when construct inner key

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 

When the outer join key of index join is not the same type as inner side, we may meet overflows error when constructing the inner join key, and it may raise errors to the client.

### What is changed and how it works?

Ignore the overflow error when constructing the inner join key.

### Check List 

Tests 

 - Unit test

Code changes

 - Has unexported function/method change

Side effects

 - None

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: executor/index_lookup_join.go [hunk 1]
```diff
@@ -509,6 +509,10 @@ func (iw *innerWorker) constructDatumLookupKey(task *lookUpJoinTask, rowIdx int)
 		}
 		innerColType := iw.rowTypes[iw.keyCols[i]]
 		innerValue, err := outerValue.ConvertTo(sc, innerColType)
+		// If the converted outerValue overflows, we don't need to lookup it.
+		if terror.ErrorEqual(err, types.ErrOverflow) {
```
Comment (winoros) [inline executor/index_lookup_join.go]: this should be a `if...else` block?

**Comment by lamxTyler** [comment]
PTAL @XuHuaiyu @zz-jason

**Pull request merged**